node/src/reactor.rs

```diff
@@ -212,6 +212,30 @@ pub trait Reactor: Sized {
     fn update_metrics(&mut self, _event_queue_handle: EventQueueHandle<Self::Event>) {}
 }
 
+/// A reactor event type.
+pub trait ReactorEvent: Send + Debug + From<ControlAnnouncement> + 'static {
+    /// Returns the event as a control announcement, if possible.
+    ///
+    /// Returns a reference to a wrapped
+    /// [`ControlAnnouncement`](`crate::effect::announcements::ControlAnnouncement`) if the event
+    /// is indeed a control announcement variant.
+    fn as_control(&self) -> Option<&ControlAnnouncement>;
+}
+
+// Special implementation for the unit type, used in unit tests.
+// TODO: Refactor unit-testing and get rid of this implementation.
+impl ReactorEvent for () {
```

Comment:
Let's feature-gate this then please if it's only for test.  Same for the `From` impl below.